R/psis.R

```diff
@@ -224,26 +234,34 @@ psis_object <-
 #' @param lr Matrix of log ratios (`-loglik`)
 #' @param r_eff Vector of relative effective sample sizes
 #' @param cores User's integer `cores` argument
+#' @param is_method Importance sampling method to use. Deafults to PSIS, but also IS and TIS are implemented.
 #' @return A list with class `"psis"` and structure described in the main doc at
 #'   the top of this file.
 #'
-do_psis <- function(log_ratios, r_eff, cores) {
+do_psis <- function(log_ratios, r_eff, cores, is_method = "PSIS") {
   stopifnot(cores == as.integer(cores))
+  stopifnot(is_method %in% implemented_is_methods())
   N <- ncol(log_ratios)
   S <- nrow(log_ratios)
   tail_len <- n_pareto(r_eff, S)
-  throw_tail_length_warnings(tail_len)
+
+  if(is_method == "PSIS") {
+    is_fun <- do_psis_i
+    throw_tail_length_warnings(tail_len)
+  }
+  if(is_method == "TIS") is_fun <- do_tis_i
+  if(is_method == "IS") is_fun <- do_is_i
```

Comment:
maybe use if else if -structure to make it clear that only one of these can happen, and error if is_method is undefined?

Reply:
Indeed! Ill fix.